### ISSUE TITLE
Docs needs to be updated!

### ISSUE THREAD

**Issue by Rouein** [owid/covid-19-data#2387]
```
SETTINGS_PATH = os.path.join(CONFIG_PATH, "gdrive_settings.yaml")
CREDENTIALS_PATH = os.path.join(CONFIG_PATH, "gdrive_credentials.json")
```
what are `gdrive_settings.yaml` and `gdrive_credentials.json`?

and why such a cat and mouse game?
`config.yaml` in `scripts` as well as `.config/cowid` as well as `.config/owid` ?

**Comment by Rouein** [comment]
as far as I undestand there are 3 file that are of importance:
1. `config.yaml` in `scripts`
2. in which you can set the path to `client_secrets.json` (`"google_credentials"`)
3. `storage.json` which is created after using `client_secrets.json`

I am lost with what is what and which is where.

**Comment by lucasrodes** [comment]
Hi @Rouein,
Thanks for raising this. 

This part of the pipeline is currently not ready to be used by the open-source community but will be soon hopefully.

> and why such a cat and mouse game?

I understand that probably this is not the most optimal way of structuring some files, but this has not been top of our priorities so far, but the collecting and accuracy of the data have been.

Again, we hope we can simplify these steps in the near future, but is not on our immediate roadmap. Thanks for your help!

**Comment by Rouein** [comment]
I understand but it is impeding the scripts development. I have spend an hour or so trying to set them to use `GSheetApi()` and still unsuccessful.
You need to implement [this](https://stackoverflow.com/questions/64783192/how-do-i-solve-a-google-sheet-api-error-in-python) for Thailand test script. I can not do it because I have no way of testing it.

**Comment by lucasrodes** [comment]
> running cowid-vax --config config.yaml asked for s3 credentials, why is there a need for aws credentials?
if it is for your use it should be in a different branch named internal or server.

We keep it in master because it has been the easiest way for the team to work properly in the last two years.

To better understand why the library is far from being optimally organized you need to put it in context and see what the goal of the library has been.

This library is a consequence of the work we do here at Our World in Data, with few resources. We have done major efforts in the last year to optimize the library and its structure, but we are not there yet.

Note that while refactoring is nice, we constantly need to decide which metrics to track, how to display them on our website, etc. Data is collected from multiple sources (more than 200 countries and several metrics) with a variety of formats; data is combined and remixed, their reliability, etc.

**Comment by lucasrodes** [comment]
Having said this, I really appreciate your input here which highlights that there is room for improvement. Thanks!

We will do an effort soon to ease the credentials and config files management of the library, and adapt the documentation accordingly. Until then, I would discourage working on scripts that need credentials.

While we aim to be fast, we need to ensure the quality of our data (our number one priority) as we transition to a new structure.

Will come back to this issue!

Thanks once again for the insights

**Comment by Rouein** [comment]
> > running cowid-vax --config config.yaml asked for s3 credentials, why is there a need for aws credentials?
> > if it is for your use it should be in a different branch named internal or server.
> 
> We keep it in master because it has been the easiest way for the team to work properly in the last two years.
> 
> To better understand why the library is far from being optimally organized you need to put it in context and see what the goal of the library has been.
> 
> This library is a consequence of the work we do here at Our World in Data, with few resources. We have done major efforts in the last year to optimize the library and its structure, but we are not there yet.
> 
> Note that while refactoring is nice, we constantly need to decide which metrics to track, how to display them on our website, etc. Data is collected from multiple sources (more than 200 countries and several metrics) with a variety of formats; data is combined and remixed, their reliability, etc.

Fully understandable! but that's why we have the open source community. You make a list of priorities and the community will contribute to get them done but that needs a sound framework so that they can tests their work. 
And that has been great so far but something as simple as using a credentials file shouldn't be the impeding factor.
A better approach in my experience has been developing your library in this case `utils.gdrive` in  different branch and if it passes the tests on a neutral platform such as a docker env then you push it to main or master.

and please do not misunderstand me, in layman's terms I'm saying: tell me what you need, I am here to help.

**Comment by lucasrodes** [comment]
Thanks once again for elaborating! Bery informative, and you are right. I'll see if I can ease credentials management this month hopefully.
